packages/snippets/bin/extract.dart

```diff
@@ -0,0 +1,142 @@
+    negatable: false,
+    help: 'Prints help documentation for this command',
+  );
+
+  final ArgResults args = parser.parse(argList);
+
+  if (args[_kHelpOption] as bool) {
+    stderr.writeln(parser.usage);
+    exit(0);
+  }
+
+  if (args[_kInputOption] == null || (args[_kInputOption] as String).isEmpty) {
+    stderr.writeln(parser.usage);
+    errorExit('The --$_kInputOption option must not be empty.');
+  }
+
+  if (args[_kOutputOption] == null || (args[_kOutputOption] as String).isEmpty) {
+    stderr.writeln(parser.usage);
+    errorExit('The --$_kOutputOption option must be specified, and not empty.');
+  }
+
+  final File input = filesystem.file(args['input'] as String);
+  if (!input.existsSync()) {
+    errorExit('The input file ${input.absolute.path} does not exist.');
+  }
+  final String flutterSource = path.join(
+    flutterInformation.getFlutterRoot().absolute.path,
+    'packages',
+    'flutter',
+    'lib',
+    'src',
+  );
+  if (!path.isWithin(flutterSource, input.absolute.path)) {
+    errorExit('Input file must be under the $flutterSource directory: ${input.absolute.path} is not.');
+  }
+
+  final Iterable<SourceElement> fileElements = getFileElements(input);
+  final SnippetDartdocParser dartdocParser = SnippetDartdocParser();
+  final SnippetGenerator snippetGenerator = SnippetGenerator();
+  dartdocParser.parseFromComments(fileElements);
+  dartdocParser.parseAndAddAssumptions(fileElements, input, silent: true);
+
+  final String srcPath = path.relative(input.absolute.path, from: flutterSource);
+  final String dstPath = path.join(
+    flutterInformation.getFlutterRoot().absolute.path,
+    'examples',
+    'api',
+  );
+  for (final SourceElement element in fileElements.where((SourceElement element) {
+    return element.sampleCount > 0;
+  })) {
+    for (final CodeSample sample in element.samples) {
+      // Ignore anything else, because those are not full apps.
+      if (sample.type != 'dartpad' && sample.type != 'sample') {
+        continue;
+      }
+      final String relativePath = path.relative(sample.start.file!.path,
+          from: flutterInformation.getFlutterRoot().absolute.path);
+      snippetGenerator.generateCode(
+        sample,
+        includeAssumptions: false,
+        copyright: _kCopyrightNotice,
+        description: 'See description in the comments in the file:\n  $relativePath',
+      );
+      final File outputFile = filesystem.file(
+        path.joinAll(<String>[
+          dstPath,
+          'lib',
+          path.withoutExtension(srcPath), // e.g. material/app_bar
+          <String>[
+            if (element.className.isNotEmpty) element.className.snakeCase,
+            element.name.snakeCase,
+            sample.index.toString(),
+            'dart',
+          ].join('.'),
+        ]),
+      );
+      await outputFile.absolute.parent.create(recursive: true);
+      if (outputFile.existsSync()) {
+        print('File $outputFile already exists!');
+        exit(-1);
```

Comment:
Would using `errorExit` be better here, or is that just for arg parsing?

Reply:
Yeah, maybe it would.